src/cpp/core/system/PosixLibraryLoader.cpp

```diff
@@ -53,6 +53,32 @@ Error loadLibrary(const std::string& libPath, void** ppLib)
    }
 }
 
+/**
+ * Verify that we can load a library (without actually loading it)
+ * by passing RTLD_NOLOAD to dlopen
+ */
+Error verifyLibrary(const std::string& libPath, void** ppLib)
```

Comment:
Since you never use `ppLib`, I don't think you should call it here. You're also never closing the handle on success, but I'm not sure if that matters.